goroutine/gid_tinygo.go

```diff
@@ -0,0 +1,37 @@
+//go:build tinygo
+
+package goroutine
+
+import (
+	"sync"
+	"unsafe"
+)
+
+const IsTinyGo = true
+
+var (
+	counter int64
+	mapping = make(map[uintptr]int64)
+	// TinyGo at the moment is single threaded so this is not needed but it's good to have anyway
+	// in case that changes. It does had ~5ns (from 20ns vs 4ns big go) but it's better to be correct.
+	// In theory the mutex could be noop on platforms where everything is single threaded.
+	lock sync.Mutex
+)
+
+func ID() int64 {
+	task := uintptr(currentTask())
+	lock.Lock() // explicit minimal critical section without using defer, on purpose.
+	if id, ok := mapping[task]; ok {
+		lock.Unlock()
+		return id
+	}
+	counter++
+	mapping[task] = counter
+	lock.Unlock()
+	return counter
+	// or, super fast but ugly large numbers/pointers:
+	//return int64(uintptr(currentTask()))
+}
+
+//go:linkname currentTask internal/task.Current
```

Comment:
I don't think I ever face this directive

Can you explain?

This comment is non blocking of course

Reply:
it means create the function `currentTask` here but when it's called, call `internal/task.Current` instead

in this case (tinygo)

https://github.com/tinygo-org/tinygo/blob/v0.32.0/src/internal/task/task_stack.go#L39

which being internal... isn't otherwise reachable

go folks are trying to get the cat back in the bag in https://github.com/golang/go/issues/67401